milvus/utils/Format.ts

```diff
@@ -467,6 +471,13 @@ export const buildDynamicRow = (
     }
   }
 
+  // do this last to prevent any non-deterministic behavior with loop order
+  if (originRow[dynamicFieldName]) {
+    // extend the dynamic field object with the input dynamic field object
+    row[dynamicFieldName] = row[dynamicFieldName] || ({} as JSON);
+    Object.assign(row[dynamicFieldName] as JSON, originRow[dynamicFieldName]);
+  }
+
```

Comment:
I'm not sure if this is the intended behaviour when the input data is something like:

```js
{
    $meta: {
        key2: "value22",
    },
    key1: "value1",
    key2: "value2",
}
```

where `key2` is a dynamic field not specified in the schema.